shared/redux/rtk-apis/enrollments/enrollments.api.ts

```diff
@@ -1,13 +1,35 @@
 import { projectApi } from "../api.config";
-import { IEnrollment } from "./enrollments.types";
+import { IEnrollment, IStudent } from "./enrollments.types";
 
 const enrollmentApi = projectApi.injectEndpoints({
   endpoints: (builder) => ({
     getClassroomStudents: builder.query<IEnrollment[], number>({
       query: (classroomId) => `enrollments/students/${classroomId}`,
+      providesTags: ["Enrollments"],
+    }),
+    getUnenrolledStudents: builder.query<IStudent[], { classroomId: number }>({
+      query: ({ classroomId }) => ({
+        url: `students/unenrolled/${classroomId}`,
+      }),
+      providesTags: ["UnenrolledStudents"],
+    }),
+    enrollStudent: builder.mutation<void, { studentId: number; classroomId: number }>({
+      query: ({ studentId, classroomId }) => ({
+        url: `/enrollments`,
```

Comment:
For tech debt ticket:
- the endpoint can be something like POST `classrooms/classroomId/enroll`